app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,78 @@
+package de.westnordost.streetcomplete.quests.opening_hours
+
+import android.util.Log
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.DateUtil
+
+class ResurveyOpeningHours(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<OpeningHoursAnswer> {
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        return arrayOf(tags["name"]!!)
+    }
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
+                }
+            }
+
+        }
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val groupName = ".old_opening_hours"
+        val nodeGroupName = groupName + "_nodes"
+        val wayGroupName = groupName + "_ways"
+        val relationGroupName = groupName + "_relations"
+        val reviewIntervalInDays = 380
+        return bbox.toGlobalOverpassBBox() + "\n" +
+                getQueryPart("node", nodeGroupName, reviewIntervalInDays) +
+                getQueryPart("way", wayGroupName, reviewIntervalInDays) +
+                getQueryPart("relation", relationGroupName, reviewIntervalInDays) +
```

Comment:
Couldn't you use `nwr`?

Reply:
I thought that `nwr` is unavailable, but I turned out to be mistaken. Thanks, this should make query significantly more readable.